metrics_handler/main.py

```diff
@@ -206,59 +112,38 @@ def get_metrics_from_events_dir(self):
     """
     tags_to_ignore = set(
         self.metric_collection_config.get('tags_to_ignore', []))
-
-    em = event_multiplexer.EventMultiplexer()
-    em.AddRunsFromDirectory(self.events_dir)
-    em.Reload()
-
-    # First pass: collect the values for each metric.
-    raw_metrics = collections.defaultdict(list)
-    for run, tags in em.Runs().items():
-      # 'Old-style' runs have a simple format and store values directly.
-      for tag in tags['scalars']:
-        if tag in tags_to_ignore:
-          continue
-        raw_metrics[tag].extend(
-            [self.MetricPoint(metric_value=x.value, wall_time=x.wall_time)
-            for x in em.Scalars(run, tag)])
-      # 'New-style' runs stores values inside of Tensor protos.
-      for tag in tags['tensors']:
-        if tag in tags_to_ignore:
-          continue
-        for t in em.Tensors(run, tag):
-          tensor_dtype = tf.dtypes.as_dtype(t.tensor_proto.dtype)
-          try:
-            val = np.frombuffer(
-                t.tensor_proto.tensor_content,
-                tensor_dtype.as_numpy_dtype).tolist()
-            assert len(val) == 1  # There should be 1 value per tensor.
-            raw_metrics[tag].append(
-                self.MetricPoint(metric_value=val[0], wall_time=t.wall_time))
-          except ValueError as e:
-            self.logger.warning(
-                'Unable to parse tag: `{}` from tensor_content: {}. '
-                'Error: {}. Consider adding this tag to tags_to_ignore '
-                'in config.'.format(tag, t.tensor_proto.tensor_content, e))
-
-    # Second pass: aggregate values for each metric based on the config.
-    final_metrics = {}
-    tag_to_custom_aggregation_strategies = self.metric_collection_config.get(
-        'metric_to_aggregation_strategies', {})
-    for tag, metrics in raw_metrics.items():
-      strategies = tag_to_custom_aggregation_strategies.get(
-          tag, self.metric_collection_config['default_aggregation_strategies'])
-      for strategy in strategies:
-        final_metrics['{}_{}'.format(tag, strategy)] = self._aggregate_metrics(
-            metrics, strategy)
-
+    raw_metrics = metrics.read_metrics_from_events_dir(
+        self.events_dir, tags_to_ignore)
+
+    default_aggregation_strategies = self.metric_collection_config.get(
+        'default_aggregation_strategies')
+    metric_to_aggregation_strategies = self.metric_collection_config.get(
+        'metric_to_aggregation_strategies')
+    final_metrics = metrics.aggregate_metrics(
```

Comment:
Right now I don't think this would translate to any alerts if it fails since it will raise a ValueError and the process will terminate. We can either pass a logger to all of the `metrics.*` methods and have those methods decide which failure are email-worthy or wrap all calls with try/catch and log the errors in this method like you do below with metrics.time_to_accuracy

Reply:
It will fall through and get caught/logged here: 

```
    try:
      logger.info('Pubsub message to process: {}'.format(msg))
      should_ack = _process_pubsub_message(msg, status_handler, logger)
    except Exception as e:
      logger.error(
          'Encountered exception `{}` while attempting to '
          'process message. The message will be acknowledged to prevent more '
          'crashes. The message was: {}'.format(e, msg))
      should_ack = True
```

I did add a try/catch to add more context to the error before re-raising it.